M2/Macaulay2/m2/startup.m2.in

```diff
@@ -158,8 +158,9 @@ if firstTime then (
      toAbsolutePath = pth -> if pth =!= "stdio" and not isAbsolutePath pth then "/" | relativizeFilename("/", pth) else pth;
 
      copyright = (
-	  "Macaulay2, version " | version#"VERSION" | newline
-	  | "Copyright 2019 Daniel R. Grayson and Michael E. Stillman:" | newline
+	  "Macaulay2, version " |
+	  replace("^release-", "", version#"git description") | newline
+	  | "Copyright 1993-2024 Daniel R. Grayson and Michael E. Stillman:" | newline
```

Comment:
Is the year strictly necessary here? (note that none of the libraries include year)

Also, might be easier to switch to `Macaulay2 release-1.YY.MM-...` so you don't need the regex replace.

Reply:
All the GPL documentation says to include the year, e.g. from https://www.gnu.org/licenses/gpl-howto.en.html

> The copyright notice should include the year in which you finished preparing the release

I'm not sure why we include all the libraries in the notice. I haven't seen that before in other projects

Reply:
So we need to change that line every year? That seems tedious (we could also infer it from the new version scheme, but that seems over engineered ...)

I don't even understand why this has to be a part of startup.m2! A documentation page seems much more appropriate.

Reply:
Here's the full suggestion: delete that stuff from startup.m2 and make `copyright` be a command that opens `help "Copyright and license"`. Also delete the documentation page "copyright".

Reply:
Done!

I also added a short copyright blurb to the startup banner that we probably should have had all along as per the GPL:

> If the program is interactive, make it output a short notice like this when it starts in an interactive mode:
>
>   ```
>   Gnomovision version 69, Copyright (C) year name of author
>   Gnomovision comes with ABSOLUTELY NO WARRANTY; for details type `show w'.
>   This is free software, and you are welcome to redistribute it
>   under certain conditions; type `show c' for details.
>  ```

R and Octave, both of which are also GPL, both have something like this.  So this is the new proposal:

```m2
Macaulay2, version 1.24.05-14-g5d5a2de127-dirty (version)
Copyright 1993-2024 Daniel R. Grayson and Michael E. Stillman
This program comes with ABSOLUTELY NO WARRANTY.  This is free
software, and you are welcome to redistribute it under certain
conditions; type 'copyright' for details.
with packages: ConwayPolynomials, Elimination, IntegralClosure, InverseSystems,
               Isomorphism, LLLBases, MinimalPrimes, OnlineLookup,
               PrimaryDecomposition, ReesAlgebra, Saturation, TangentCone,
               Truncations, Varieties

i1 : copyright

o1 = Copyright and license
     *********************

     Macaulay2, its object code, source code, and documentation, are copyright
     by Daniel R. Grayson and Michael E. Stillman.  We permit you to use it
     either under the terms of the GNU General Public License, version 2, as
     published by the Free Software Foundation, and as contained in the file
     "COPYING-GPL-2" accompanying the program, or under the terms of the GNU
     General Public License, version 3, as published by the Free Software
     Foundation, and as contained in the file "COPYING-GPL-3" accompanying the
     program.

     Various packages are included with Macaulay2 and come with their own
     copyright notices and licenses, see "packages provided with Macaulay2".

     Some free libraries have been compiled into (or linked with) Macaulay2, and
     some free programs, or packages of programs, with their libraries, have
     been compiled and distributed with Macaulay2:

     Menu
     ----

     libraries

       * "Singular-Factory"
       * "frobby"
       * "MPIR"
       * "GNU MP"
       * "MPFR"
       * "MPFI"
       * "GC garbage collector"
       * "LAPACK"
       * "BLAS"
       * "NTL library"
       * "FLINT"
       * "givaro"
       * "fflas-ffpack"
       * "MPSolve" -- a library for finding roots of univariate polynomials
       * "Boost" -- a collection of C++ source libraries

     programs and their libraries

       * "4ti2"
       * "gfan"
       * "normaliz"
       * "csdp"
       * "nauty"
       * "cdd+"
       * "lrslib"
       * "topcom"
       * "cohomCalg"

```
The `--copyright` command line option now just adds this new command to the list of startup functions using `addStartFunction`.

Reply:
Hmm I really dislike this actually lol

Reply:
If we must, can we either:
1. condition it to display only on the first run (or if `-q` is given)
2. make it just mention the commands like python:
```
Type "help", "copyright", "credits" or "license" for more information.
```
Admittedly they are not using GPL, but we haven't had this notice until now and nothing has gone wrong, I doubt someone will steal M2 if we don't remind them of the copyright every time it runs XD